### ISSUE TITLE
Bug with `isPOD` and generic record types

### ISSUE THREAD

**Issue by dlongnecke-cray** [chapel-lang/chapel#24172]
```chapel
use Types;

record r { type T; var x: T; }

proc main() {
  writeln(isPOD(r(?)));
}
```

```
Foo.chpl:3: error: cannot default-initialize a variable with generic type
Note: This source location is a guess.
note: '<temporary>' has generic type 'r'
note: cannot find initialization point to split-init this variable
```

In the debugger I can see that it's referring to the auto-generated deinit which has its resolution triggered by `resolveAutoCopyEtc`, called from `propagateNotPod`.

**Comment by dlongnecke-cray** [comment]
I think that we should probably have a response to whether or not a generic type is POD or not, and IMO the answer should be "no", but "print an error instead" is also an acceptable answer.

**Comment by bradcray** [comment]
It seems to me like an uninstantiated generic is neither POD nor non-POD since you can't create an instance of it without instantiating it.  E.g., in your example, `r(int)` would be true and `r(MyClass)` would be false, so who's to say what `r(?)` should mean, since it could conceivably be either?

In saying:

> IMO the answer should be "no"

is your argument that when the question can't be answered, then "no" is the right answer to give?  I agree it's better than "yes", but I think I'd rather be told that my question was a bad one (i.e., I'd like to get an error or some third result) than to accidentally ask a bad question without realizing it, get a "no" and draw the wrong conclusion as a result.  By analogy, we could similarly have `isPod()` on a procedure name return `no` as well, but if I'm making that call, I'm probably making a mistake in doing it at all?

**Comment by dlongnecke-cray** [comment]
Sounds like the error message is ultimately the way to go, then. FWIW I am torn between  `isPOD(p)` where `p` is some procedure always returning `true` (since they are just pointers), or `true` and `false` depending on whether the thing is generic, or something else (to handle closures, perhaps). But I think whatever we choose we can make it consistent such that it errors if it doesn't make sense.

**Comment by bradcray** [comment]
> since they are just pointers

Oh, that's an interesting point (ha ha).  I was actually trying to think of something rather abstract / obviously nonsensical (like I almost suggested "module name") but didn't consider that interpretation.

(Of course, today, they aren't pointers (yet), right?  Or did you complete that and I failed to notice?)

**Comment by dlongnecke-cray** [comment]
Yeah, they're not pointers yet. I agree that we can definitely find some more nonsensical things to throw at `isPOD` that it should be able to handle gracefully.
